contracts/inverse-bonds/types/Pro/Viewer.sol

```diff
@@ -0,0 +1,245 @@
+
+    constructor() ProMarketCreator() {}
+
+    /* ========== EXTERNAL VIEW ========== */
+
+    /**
+     * @notice             calculate current market price of base token in quote tokens
+   * @dev                accounts for debt and control variable decay since last deposit (vs _marketPrice())
+   * @param _id          ID of market
+   * @return             price for market in base token decimals
+   *
+   * price is derived from the equation
+   *
+   * p = c * d
+   *
+   * where
+   * p = price
+   * c = control variable
+   * d = debt
+   *
+   * d -= ( d * (dt / l) )
+   * 
+   * where
+   * dt = change in time
+   * l = length of program
+   *
+   * if price is below minimum price, minimum price is returned
+   * this is enforced on deposits by manipulating total debt (see _decay())
+   */
+    function marketPrice(uint256 _id) public view override returns (uint256) {
+        uint256 price =
+        currentControlVariable(_id)
+        * currentDebt(_id)
+        / (10 ** metadata[_id].baseDecimals);
+        return
+        (price > markets[_id].minPrice)
+        ? price
+        : markets[_id].minPrice;
+    }
+
+    /**
+     * @notice             payout due for amount of quote tokens
+   * @dev                accounts for debt and control variable decay so it is up to date
+   * @param _amount      amount of quote tokens to spend
+   * @param _id          ID of market
+   * @return             amount of base tokens to be paid
+   */
+    function payoutFor(uint256 _amount, uint256 _id) public view override returns (uint256) {
+        Metadata memory meta = metadata[_id];
+        return
+        _amount
+        * 10 ** (2 * meta.baseDecimals)
+        / marketPrice(_id)
+        / 10 ** meta.quoteDecimals;
+    }
+
+    /**
+     * @notice             calculate debt factoring in decay
+   * @dev                accounts for debt decay since last deposit
+   * @param _id          ID of market
+   * @return             current debt for market in base token decimals
+   */
+    function currentDebt(uint256 _id) public view override returns (uint256) {
+        uint256 decay = markets[_id].totalDebt
+        * (block.timestamp - metadata[_id].lastDecay)
+        / metadata[_id].length;
+        return markets[_id].totalDebt - decay;
+    }
+
+    /**
+     * @notice             up to date control variable
+   * @dev                accounts for control variable adjustment
+   * @param _id          ID of market
+   * @return             control variable for market in base token decimals
+   */
+    function currentControlVariable(uint256 _id) public view returns (uint256) {
+        (uint256 decay,,) = _controlDecay(_id);
+        return terms[_id].controlVariable - decay;
+    }
+
+    /**
+     * @notice             returns maximum quote token in for market
+   */
+    function maxIn(uint256 _id) public view returns (uint256) {
+        Metadata memory meta = metadata[_id];
+        return
+        markets[_id].maxPayout
```

Comment:
This is also something I wonder if the `maxCapacity` should be considered as well because if the capacity goes down to something lower than `maxPayout`, what this returns is true for the `maxPayout` but not true for the "market" itself.

Maybe something like:

```
Math.min(
  markets[_id].maxPayout,
  markets[_id].maxCapacity,
)
```